### PR TITLE
Log history

### PR DESCRIPTION
**Pull request by ricardopereira**
Regarding #596.

### CODE REVIEW

File: Source/ARTLog.m [hunk 1]
```diff
@@ -43,23 +43,54 @@ - (NSString *)toString {
     return [NSString stringWithFormat:@"%s: %@", logLevelName(self.level), self.message];
 }
 
+- (NSString *)description {
+    return [self toString];
+}
+
+#pragma mark - NSCoding
```
Comment (tcard) [inline Source/ARTLog.m]: What is this for?

Comment (ricardopereira) [inline Source/ARTLog.m]: To save the object in disk with `NSUserDefaults`.

Comment (ricardopereira) [inline Source/ARTLog.m]: If necessary. Currently, the logger does not store the history.

File: Source/ARTLog.m [hunk 2]
```diff
@@ -43,23 +43,54 @@ - (NSString *)toString {
...
+    NSUInteger _historyLines;
 }
 
 - (instancetype)init {
-    return [self initCapturingOutput:false];
+    return [self initCapturingOutput:true];
 }
 
 - (instancetype)initCapturingOutput:(BOOL)capturing {
+    return [self initCapturingOutput:true historyLines:10];
```
Comment (mattheworiordan) [inline Source/ARTLog.m]: 10 seems awfully low, should we bump to say 50 or 100?

Comment (ricardopereira) [inline Source/ARTLog.m]: Done c3afa25.

**Pull request merged**